src/app/Http/Controllers/Admin/SettingController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Http\Controllers\Admin;
+
+use App\Http\Resources\Admin\SettingResource;
+use App\Interfaces\SettingRepositoryInterface;
+use App\Models\Setting;
+use Illuminate\Http\Request;
+
+class SettingController extends ResourceController
+{
+    private SettingRepositoryInterface $settingRepository;
+
+    public function __construct(SettingRepositoryInterface $settingRepository)
+    {
+        $this->settingRepository = $settingRepository;
+    }
+
+    public function index(Request $request)
+    {
+        $this->authorize('viewAny', Setting::class);
```

Comment:
You should be able to call `$this->authorizeResource` in the controller rather than having to call `$this->authorize` in each endpoint. See the other admin api controllers.

Reply:
ok cool ill check that out